packages/mui-base/src/NumberField/Root/NumberFieldRoot.types.ts

```diff
@@ -121,8 +121,9 @@ export interface UseNumberFieldRootParameters {
   /**
    * Callback fired when the number value changes.
    * @param {number | null} value The new value.
+   * @param {Event} event The event that triggered the change.
    */
-  onChange?: (value: number | null) => void;
+  onValueChange?: (value: number | null, event?: Event) => void;
```

Comment:
Why do you use native events instead of React synthetic ones?

Reply:
Some of them are native, so I decided to just always pass the native one

Reply:
I think people are used to work with React-wrapped events. It's easy to get a native event from a synthetic one, but impossible to go the other way around. I think we could list all the possible event types they can expect (like `React.MouseEvent | React.KeyboardEvent | WheelEvent`)

Reply:
The event passed to Tooltip/Popover are always native however, since that's what Floating UI does. 

I think making it consistently the native one is fine, as no one has requested the React synthetic event in that library. 

Moreover, we may change the implementation internally from a React event to one using `.addEventListener()` (there are sometimes rare bugs with synthetic events), which will break if we don't pass the native one

Reply:
OK, if it's consistent across all the components, then I'm fine with it.